sdk/core/Azure.Core/src/Serialization/ObjectSerializer.cs

```diff
@@ -31,6 +32,42 @@ public abstract class ObjectSerializer
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during serialization.</param>
         public abstract ValueTask SerializeAsync(Stream stream, object? value, Type inputType, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Convert the provided value to it's binary representation and return it as a <see cref="BinaryData"/> instance.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="inputType">The type of the <paramref name="value"/> to convert.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during serialization.</param>
+        /// <returns>The object's binary representation as <see cref="BinaryData"/>.</returns>
+        public BinaryData SerializeToBinaryData(object? value, Type? inputType = default, CancellationToken cancellationToken = default) =>
+            SerializeToBinaryDataInternalAsync(value, inputType, false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Convert the provided value to it's binary representation and return it as a <see cref="BinaryData"/> instance.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="inputType">The type of the <paramref name="value"/> to convert.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during serialization.</param>
+        /// <returns>The object's binary representation as <see cref="BinaryData"/>.</returns>
+        public async ValueTask<BinaryData> SerializeToBinaryDataAsync(object? value, Type? inputType = default, CancellationToken cancellationToken = default) =>
+            await SerializeToBinaryDataInternalAsync(value, inputType, true, cancellationToken).ConfigureAwait(false);
+
+        private async ValueTask<BinaryData> SerializeToBinaryDataInternalAsync(object? value, Type? inputType, bool async, CancellationToken cancellationToken)
+        {
+            using var stream = new MemoryStream();
+            inputType ??= value?.GetType() ?? typeof(object);
```

Comment:
isn't SerializeAsync already doing this?

Reply:
No, because these are required non-nullable parameters in ObjectSerializer.Serialize.